lib/fingerprint.js

```diff
@@ -51,6 +53,21 @@ module.exports = function(opt, hash_provider) {
             return cb();
         }
 
+        // test if the url should be hashed or not
+        if (opt.exclude) {
+            excluded = opt.exclude.some(function (rule) {
+                if (typeof rule === 'string' && orig.indexOf(rule) !== -1) {
+                    return true;
+                } else if (typeof rule === 'object' && rule.test && rule.test(orig)) {
+                    return true;
+                }
+            });
+        }
+
+        if (excluded) {
```

Comment:
You don't need this if statement again here and can just check excluded above in the previous if